x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/EnrichCache.java

```diff
@@ -52,12 +47,13 @@
  */
 public final class EnrichCache {
 
+    private static final CacheValue EMPTY_CACHE_VALUE = new CacheValue(List.of(), 0L);
```

Comment:
This is currently not accurate/correct, as the occupied heap space is not `0` for an empty cache result. I think we can either use a static size or compute the size. The static value won't be completely accurate because the size of the cache key is dependent on the size of the value `Object` used in there, but it will allow us to keep this static `EMPTY_CACHE_VALUE` which in turn will save us a significant amount of heap space.